cl_dll/hud_oldscoreboard.cpp

```diff
@@ -0,0 +1,309 @@
+				break;
+
+			xpos = NAME_RANGE_MIN + xpos_rel;
+
+			char szTeamName[128];
+			if (scoreboard->m_iIsATeam[iRow] == TEAM_SPECTATORS)
+			{
+				r = g = b = 100;
+				snprintf(szTeamName, ARRAYSIZE(szTeamName), "%s", CHudTextMessage::BufferedLocaliseTextString( "#Spectators"));
+			}
+			else
+			{
+				r = iTeamColors[team_info->teamnumber % iNumberOfTeamColors][0];
+				g = iTeamColors[team_info->teamnumber % iNumberOfTeamColors][1];
+				b = iTeamColors[team_info->teamnumber % iNumberOfTeamColors][2];
+				snprintf(szTeamName, ARRAYSIZE(szTeamName), "%s", team_info->name);
+			}
+
+			// Cut off the name if it'd overlap score
+			// - 5 to allow for some more space between the name and kills
+			while ( gHUD.GetHudStringWidth(szTeamName) + nameoffset + NAME_RANGE_MIN > KILLS_RANGE_MIN - 5 )
+			{
+				szTeamName[strlen(szTeamName) - 1] = '\0';
+			}
+			// draw their name (left to right)
+			gHUD.DrawHudString(xpos, ypos, 0, szTeamName, r, g, b);
+
+			// draw kills (right to left)
+			xpos = KILLS_RANGE_MAX + xpos_rel;
+			gHUD.DrawHudNumberStringFixed(xpos, ypos, team_info->frags, r, g, b);
+
+			// draw divider
+			xpos = DIVIDER_POS + xpos_rel;
+			gHUD.DrawHudStringCentered(xpos, ypos, "/", r, g, b);
+
+			// draw deaths
+			xpos = DEATHS_RANGE_MAX + xpos_rel;
+			gHUD.DrawHudNumberStringFixed(xpos, ypos, team_info->deaths, r, g, b);
+			list_slot++;
+		}
+		else
+		{
+			// Draw player info line
+			if (gHUD.m_Teamplay)
+				nameoffset = 10;
+
+			hud_player_info_t* pl_info = &g_PlayerInfoList[sorted];
+			extra_player_info_t* pl_info_extra = &g_PlayerExtraInfo[sorted];
+
+			ypos = ROW_TOP + ROW_RANGE_MIN + (list_slot * ROW_GAP);
+
+			// check we haven't drawn too far down
+			if ( ypos > ROW_RANGE_MAX )  // don't draw to close to the lower border
+				break;
+
+			r = iTeamColors[pl_info_extra->teamnumber % iNumberOfTeamColors][0];
+			g = iTeamColors[pl_info_extra->teamnumber % iNumberOfTeamColors][1];
+			b = iTeamColors[pl_info_extra->teamnumber % iNumberOfTeamColors][2];
+
+			xpos = NAME_RANGE_MIN + xpos_rel; // Set xpos for name here so that the flag is at the correct position
+
+			// If this player is carrying a CTF flag, draw the sprite for it
+			if (gHUD.m_CTF.GetBlueFlagPlayerIndex() == sorted || gHUD.m_CTF.GetRedFlagPlayerIndex() == sorted)
+			{
+				SPR_Set(m_IconFlagScore.spr, 200, 200, 200);
+				SPR_DrawAdditive(0, xpos - 10, ypos + 5, &m_IconFlagScore.rc);
+			}
+
+			if (pl_info->thisplayer) // if it's local player's name, draw it with a different color
+			{
+				r = g = b = 255;
+				// overlay the background in blue, then draw the score text over it
+				FillRGBA( NAME_RANGE_MIN + xpos_rel - 5, ypos + 5, PING_RANGE_MAX - 5, ROW_GAP, 0, 0, 255, 70);
+			}
+
+			char szName[128];
+			char colorlessName[128];
+			int specoffset = 0;
+			snprintf(szName, ARRAYSIZE(szName), "%s", pl_info->name);
+			color_tags::strip_color_tags(colorlessName, szName, ARRAYSIZE(colorlessName));
```

Comment:
Maybe we want to draw names in color?

Reply:
From the PR text:
> No support for color tags as I couldn't be bothered personally (because of the nickname cutoff code & I already spent way too much time than I anticipated on this) and they weren't in the original either.

The names use hud_color and in TDM modes team color.

I had it in mind as a toggleable cvar (I know, I know), since I from what I've gathered this feature is for the hardcore AG 6.6 (Alex comes in mind) players who want the legacy stuff and the old scoreboard never did support colortags (nor does the VGUI scorepanel). 

I then removed it altogether since I couldn't be bothered to make it work for both, since it requires not-as-trivial-as-is-right-now solution to cutting off the nicknames.

I could (except for the `(S)` part) make it work with just colortags always enabled, but then in TDM modes the "white" (non-colortagged) letters will be the team font which breaks the "color tag" principle. (Since it should be the hud_color font) - e.g. really^2LongNickname

Anyway, if you really think colortags are a priority, I'll go ahead and implement them, but personally I'd still rather cvar it.

Here's an older screenshot (from like 25th of April, obviously the positions are all off etc.) with color tags for illustration of awfulness (IMO, anyway):

![crossfire0065](https://user-images.githubusercontent.com/5108747/119550338-95276e00-bd98-11eb-9cb0-688d01618747.png)

Reply:
Disregard (almost) everything I said, I am retarded & didn't realize there's already a `GetHudStringWidthWithColorTags` function which makes this easier. I still insist on this being a cvar based on the arguments I provided, pushing a version with a cvar and immediately asking:

Is this OK or can this blow up?
```
color_tags::strip_color_tags(szName, szName, ARRAYSIZE(szName));
```

Reply:
Hm, could you ask ppl who want the feature if they want color tags or no color tags? I think it makes the most sense to have only the behavior they want, no need for a toggle.

Reply:
Never mind, colortags in oldscoreboard are default in AG 6.6, I just misremembered, sorry. Smiley and Alex said to leave them in by default like in AG 6.6, so I've removed the cvar as per this discussion. :)